Src/WebPrac/Controllers/Product2Controller.cs

```diff
@@ -61,8 +63,19 @@ public JsonResult DeleteProduct(int pid)
         }
         
         [HttpPost]
-        public JsonResult Save(ProductDTO dto)
+        public ActionResult Save(ProductDTO dto)
         {
+	        if (dto.Name.IsEmpty() || Convert.ToString(dto.Price, CultureInfo.InvariantCulture).IsEmpty())
```

Comment:
FYI. We can use annotation on our properties in Class with error message and here we can just call model.is valid prperty.